src/ext/pagination.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime
+from math import ceil
+from typing import Tuple
+
+from nextcord import ButtonStyle, Colour, Embed, Interaction, ui
+
+
+class PaginationView(ui.View):
+    current_page: int = 1
+
+    def __init__(
+        self,
+        data: list[Tuple] = [],
+        *,
+        title: str | None = "Not specified",
+        description: str | None = None,
+        separator: int = 5,
+        icon_url: str
+        | None = "https://lh3.googleusercontent.com/drive-viewer/AITFw-z_IgjwziybK0dE7JNu5i92UEkthC_H89iGhNIiUN_i1cuEnYbi9ijZPBnGRsOGNm465Rzu_GkviBlhjxwAXEp8uNGKxg=w3024-h1514",
+        color: Colour | None = Colour(0xADD8E6),
+        timeout: float | None = 180,
+        auto_defer: bool = True,
+    ) -> None:
+        super().__init__(timeout=timeout, auto_defer=auto_defer)
+        self.data = data
+        self.title = title
+        self.description = description
+        self.separator = separator
+        self.icon_url = icon_url
+        self.color = color
+
+    @property
+    def last_page(self):
+        if not self.data:
+            return 1
+        return ceil(len(self.data) / self.separator)
+
+    @property
+    def until_item(self):
+        return self.current_page * self.separator
+
+    @property
+    def from_item(self):
+        return self.until_item - self.separator
+
+    async def send(self, interaction: Interaction):
+        self.message = await interaction.send(view=self)
+        await self.update_message(self.data[: self.separator])
```

Comment:
why use `.send` method on view when you can attatch it as `view` kwarg when sending